eslint.config.mjs

```diff
@@ -65,6 +69,12 @@ export default tsEslint.config({
     "sonarjs/no-element-overwrite": "error",
     "sonarjs/no-identical-conditions": "error",
     "sonarjs/no-identical-expressions": "error",
+    "sonarjs/new-cap": "off",
+    "sonarjs/different-types-comparison": "off",
+    "sonarjs/sonar-prefer-regexp-exec": "off",
+    "sonarjs/function-return-type": "off",
+    "sonarjs/no-misleading-array-reverse": "off",
+    "sonarjs/slow-regex": "off",
```

Comment:
All new rules?

Reply:
Due to `sonar` update they introduced news rules that were annoying (in my opinion) so I disabled them.

Reply:
Something feels off about having to manually disable rules. I understand normally it's an opt in list. I don't understand why new rules would have been by default applied. 

Anyways I looked these up and I'm fine to disable them.

Reply:
Some of them felt bad, like `function-return-type` not letting you return `A | B`, or slow regex that would literally forbid any regex that use forward lookup etc. Most of these were warnings not errors but they cluttered the output in my opinion.